tests/datadriven/test_reads.py

```diff
@@ -24,6 +29,76 @@ def getDataModelClass(self):
     def getProtocolClass(self):
         return protocol.GAReadGroupSet
 
-    def testFixMe(self):
+    def testGetReadAlignmentsBothRefs(self):
+        # test that querying by both referenceName and referenceId fails
+        with self.assertRaises(exceptions.BadReadsSearchRequestBothRefs):
+            readGroupSet = self._gaObject
+            for readGroup in readGroupSet._readGroups:
+                list(readGroup.getReadAlignments("a", 5, 0, None))
+
+    def testGetReadAlignmentsByRef(self):
+        # test that getReadAlignments works when querying by either
+        # referenceName or referenceId
+        readGroupSet = self._gaObject
+        for readGroup in readGroupSet._readGroups:
+            self.resetSamFile(readGroup)
+            refNames = collections.defaultdict(int)
+            refIds = collections.defaultdict(int)
+            for read in readGroup._samFile:
```

Comment:
The gaObjects should be treated as black boxes here. We'll need to open an independent SamFile for each of the readgroups here in this class. See the distinction between pyVCF and GA variants in the `test_variants` module. Ideally we'd like to use another library to get at the contents of the alignment files, but this probably isn't feasible.